server/services/ai.py

```diff
@@ -214,22 +214,66 @@ def _delexicalize_query(query: str,
     return ''.join(output)
 
 
+_KEY_NORMALIZATION = {
+    "measurementDenominator": "md",
+    "measurementQualifier": "mq",
+    "measuredProperty": "mp",
+    "populationType": "pt",
+    "statType": "st",
+}
+
+
+def _build_query(
+        query: str,
+        key_values: Iterator[Tuple[str, str]]) -> Iterator[Tuple[str, str]]:
+    terms = []
+    for token in query.split():
+        terms.append(f'sn:"{token}"')
+    for key, value in key_values:
+        key = _KEY_NORMALIZATION.get(key, key)
```

Comment:
should the default "key" be ""? rather than the key?

Reply:
No this is correct. We normalize only if we have a normalization entry, otherwise the key is considered valid, as it can be a constrained property key.